### PR TITLE
chore(js): converts more scripts to AMD and inline require()

### PR DESCRIPTION
**Pull request by mrclay**

### CODE REVIEW

File: mod/profile/views/default/profile/js.php [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-	}
-}, 400);
+
+require(['jquery', 'elgg'], function ($, elgg) {
+	// force the first column to at least be as large as the profile box in cols 2 and 3
+	// we also want to run before the widget init happens so priority is < 500
+	elgg.register_hook_handler('init', 'system', function() {
+		// only do this on the profile page's widget canvas.
+		if ($('.profile').length) {
+			$('#elgg-widget-col-1').css('min-height', $('.profile').outerHeight(true) + 1);
```
Comment (hypeJunction) [inline mod/profile/views/default/profile/js.php]: We need to add a css rule to bypass this on mobile devices

File: mod/discussions/views/default/elgg/discussion.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		});
+	};
+
+	elgg.register_hook_handler('init', 'system', elgg.discussion.init);
+
+	/**
+	 * elgg.discussion.Reply object is deprecated. Do not call it directly.
+	 * @deprecated 2.3
+	 */
+	elgg.discussion.Reply = Reply;
```
Comment (hypeJunction) [inline mod/discussions/views/default/elgg/discussion.js]: Return elgg.discussion?

File: mod/likes/start.php [hunk 3]
```diff
@@ -14,7 +14,10 @@
 function likes_init() {
 
 	elgg_extend_view('elgg.css', 'likes/css');
-	elgg_extend_view('elgg.js', 'likes/js');
+
+	// inline module
+	elgg_extend_view('elgg.js', 'elgg/likes.js');
+	elgg_require_js('elgg/likes');
```
Comment (hypeJunction) [inline mod/likes/start.php]: Ok for BC. In 3.0 should be required by menu items via `deps`

File: mod/messageboard/start.php [hunk 4]
```diff
@@ -11,8 +11,9 @@
  * MessageBoard initialisation
  */
 function messageboard_init() {
-	// js
-	elgg_extend_view('elgg.js', 'messageboard/js');
+	// inline module
+	elgg_extend_view('elgg.js', 'elgg/messageboard.js');
+	elgg_require_js('elgg/messageboard');
```
Comment (hypeJunction) [inline mod/messageboard/start.php]: Do we need this on all requests?

File: mod/messages/views/default/messages/js.php [hunk 5]
```diff
@@ -1,7 +1,9 @@
 //<script>
-// messages plugin toggle
-elgg.register_hook_handler('init', 'system', function() {
-	$("#messages-toggle").click(function() {
-		$('input[type=checkbox]').click();
+
```
Comment (hypeJunction) [inline mod/messages/views/default/messages/js.php]: Deptecate the view by a proper AMD?

File: mod/profile/views/default/profile/js.php [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
-// force the first column to at least be as large as the profile box in cols 2 and 3
-// we also want to run before the widget init happens so priority is < 500
-elgg.register_hook_handler('init', 'system', function() {
-	// only do this on the profile page's widget canvas.
-	if ($('.profile').length) {
-		$('#elgg-widget-col-1').css('min-height', $('.profile').outerHeight(true) + 1);
-	}
-}, 400);
+
+require(['jquery', 'elgg'], function ($, elgg) {
```
Comment (hypeJunction) [inline mod/profile/views/default/profile/js.php]: This might be duplicating code from elgg/widgets

**Comment by hypeJunction** [comment]
Looks fine but haven't tested

File: mod/discussions/views/default/elgg/discussion.js (hunk 2 above)
Comment (mrclay) [inline mod/discussions/views/default/elgg/discussion.js]: Is returning an API really necessary? It's stuff we then have to support throughout 3.x.

File: mod/messageboard/start.php (hunk 4 above)
Comment (mrclay) [inline mod/messageboard/start.php]: Probably not; I wanted to keep this conversion simple.

File: mod/discussions/views/default/elgg/discussion.js (hunk 2 above)
Comment (hypeJunction) [inline mod/discussions/views/default/elgg/discussion.js]: I think having access to Reply prototype might come in handy, but up to you

File: mod/messageboard/start.php (hunk 4 above)
Comment (hypeJunction) [inline mod/messageboard/start.php]: As you say, we will have to support this through 3.x. I guess the impact is minimal butwe would have to keep the module available on all requests

File: mod/profile/views/default/profile/js.php (hunk 1 above)
Comment (mrclay) [inline mod/profile/views/default/profile/js.php]: Can you add an issue for this? The goal of this PR is limited: get rid of direct $/elgg access.

File: mod/messageboard/start.php (hunk 4 above)
Comment (mrclay) [inline mod/messageboard/start.php]: This is 2.x. We can make more changes in master.

**Comment by mrclay** [comment]
Are you aware of any other direct use of globals: elgg, $, jQuery, etc? I want to get them all at least wrapped in require() here. Can always do more in later PRs.

**Comment by hypeJunction** [comment]
Discussion JS was the only one on my radar and this covers it.

File: mod/profile/views/default/profile/js.php (hunk 1 above)
Comment (hypeJunction) [inline mod/profile/views/default/profile/js.php]: There should already be one somewhere

**Pull request merged**